source/llvm/LLVMModelDataSymbols.cpp

```diff
@@ -1625,17 +1625,42 @@ size_t LLVMModelDataSymbols::getEventBufferSize(size_t eventId) const
 
 bool LLVMModelDataSymbols::isNamedSpeciesReference(const std::string& id) const
 {
-    return namedSpeciesReferenceInfo.find(id) != namedSpeciesReferenceInfo.end();
+    if (namedSpeciesReferenceInfo.find(id) != namedSpeciesReferenceInfo.end()) {
+        return true;
+    }
+    if (id.find(":") != string::npos) {
+        //LS: This whole section is for when we ask for rateOf(specid).
+        string rxnid = id.substr(0, id.find(":"));
+        string specid = id.substr(id.find(":") + 1, id.length());
```

Comment:
Is that 'id.length()' the write "length" for the rest of id characters?

Reply:
Yup; the end character is not included in substr.